packages/flutter/lib/src/rendering/sliver_list.dart

```diff
@@ -38,24 +47,171 @@ class RenderSliverList extends RenderSliverMultiBoxAdaptor {
   /// the main axis.
   RenderSliverList({
     required super.childManager,
-  });
+    bool enableItemExtentsCaching = false,
+  }) : _enableItemExtentsCaching = enableItemExtentsCaching;
+
+  /// Whether to cache the main axis extents of all laid out children.
+  ///
+  /// If true, the main axis extent of child will be cached as
+  /// index/extent pairs when the child first laid out.
+  ///
+  /// If false, the main axis extents of children will not be cached and
+  /// previously cached data will be cleared.
+  ///
+  /// The cached data will be used to efficiently determine the target children
+  /// within the lazy loading scope, which helps improve layout performance.
```

Comment:
Good point!
I considered exposing a property named like "Set<int> invalidCachedItemExtent" to let the render object delete the corresponding cached data.
But this seems to lack grace and cannot invalidate all cached values easily. Do you have any good suggestions?

Reply:
Hmm, we could rely on the SliverChildDelegate.shouldRebuild to help invalidate the cache. In the SliverChildListDelegate that would work, but in the SliverChildBuilderDelegate it would always invalidate.. Maybe @chunhtai has some thoughts on this. Particularly with the child reordering edge cases, I think @chunhtai has some familiarity. :)

Reply:
I am not sure how this would work, even if the `SliverChildListDelegate` doesn't change, the child itself can still change size.

Reply:
Oh right! That get's me every time. It is very common for folks to change size of scrollable children through animations or in response to the scroll position changing.